azure-cs-cosmosapp-component/VmScaleSets.cs

```diff
@@ -0,0 +1,293 @@
+                                    {
+                                        MetricTrigger = new AutoscaleSettingProfilesRulesMetricTriggerArgs
+                                        {
+                                            MetricName = "Percentage CPU",
+                                            MetricResourceId = scaleSet.Id,
+                                            Operator =  "GreaterThan",
+                                            Statistic = "Average",
+                                            Threshold = 75,
+                                            TimeAggregation = "Average",
+                                            TimeGrain = "PT1M",
+                                            TimeWindow = "PT5M",
+                                        },
+                                        ScaleAction= new AutoscaleSettingProfilesRulesScaleActionArgs
+                                        {
+                                            Cooldown = "PT1M",
+                                            Direction = "Increase",
+                                            Type = "ChangeCount",
+                                            Value = 1,
+                                        },
+                                    },
+                                    new AutoscaleSettingProfilesRulesArgs
+                                    {
+                                        MetricTrigger = new AutoscaleSettingProfilesRulesMetricTriggerArgs
+                                        {
+                                            MetricName = "Percentage CPU",
+                                            MetricResourceId = scaleSet.Id,
+                                            Operator =  "LessThan",
+                                            Statistic = "Average",
+                                            Threshold = 25,
+                                            TimeAggregation = "Average",
+                                            TimeGrain = "PT1M",
+                                            TimeWindow = "PT5M",
+                                        },
+                                        ScaleAction= new AutoscaleSettingProfilesRulesScaleActionArgs
+                                        {
+                                            Cooldown = "PT1M",
+                                            Direction = "Decrease",
+                                            Type = "ChangeCount",
+                                            Value = 1,
+                                        },
+                                    },
+                                }
+                            }
+                        },
+                        TargetResourceId = scaleSet.Id,
+                    },
+                    options);
+
+                return new AzureEndpoint(publicIp.Id);
+            };
+
+        }
+    }
+}
```

Comment:
File is quite big, maybe we can break it up further :)

Reply:
I'll keep the same structure as the TS one for now